### PR TITLE
DOC: Clear some warnings for daal4py doc builds

### PR DESCRIPTION
**Pull request by david-cortes-intel**
## Description

This PR resolves many warnings that pop out when building the `daal4py` docs and which prevent the CI jobs from building those docs correctly.

Note that it doesn't solve all of the warnings - after this PR, one would still see many of these cases:
```
scikit-learn-intelex/doc/daal4py/docstring of daal4py._daal4py.classifier_prediction_result:1: WARNING: duplicate object description of daal4py.classifier_prediction_result, other instance in algorithms, use :noindex: for one of them
```

(I'm not sure what to do about those)

---

PR should start as a draft, then move to ready for review state after CI is passed and all applicable checkboxes are closed.
This approach ensures that reviewers don't spend extra time asking for regular requirements.

You can remove a checkbox as not applicable only if it doesn't relate to this PR in any way.
For example, PR with docs update doesn't require checkboxes for performance while PR with any change in actual code should have checkboxes and justify how this code change is expected to affect performance (or justification should be self-evident).

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.

**Performance**

Not applicable.

### CODE REVIEW

**Comment by david-cortes-intel** [comment]
This worked for a local build, but it seems it was not enough for the CI job named "Docs":
https://dev.azure.com/daal/daal4py/_build/results?buildId=40827&view=logs&j=7e620c85-24a8-5ffa-8b1f-642bc9b1fc36&t=dadee17e-671e-5fd8-4599-c783ec4d74d0

It still fails to import `daal4py` during the doc build step. Wonder if the issue might be that it is not building it.

File: doc/daal4py/conf.py [hunk 1]
```diff
@@ -29,9 +29,10 @@
 # add these directories to sys.path here. If the directory is relative to the
 # documentation root, use os.path.abspath to make it absolute, like shown here.
 #
-# import os
-# import sys
-# sys.path.insert(0, os.path.abspath('.'))
+import os
+import sys
+
+sys.path.insert(0, os.path.abspath("../.."))
```
Comment (icfaust) [inline doc/daal4py/conf.py:35]: Since I don't know this side, but the PR looks simple enough, small explanation here as a comment and I'll approve (given it makes sense).

Comment (david-cortes-intel) [reply doc/daal4py/conf.py:35]: The docs are generated from the python docstrings through `autodoc`. That means it needs to be able to import the package in order to build the docs.

This line adds the folder where `daal4py` would be to the paths where python searches, assuming you built the extension beforehand. Looks like it's not working in the CI job "docs" though, but the same logic is used in sklearnex, and works if you try `make html` locally:
https://github.com/uxlfoundation/scikit-learn-intelex/blob/95bd1eabcca2e4d0565423be1dba3b58deaf8901/doc/sources/conf.py#L35

**Review by icfaust — APPROVED** [review]
